src/unix/fs.c

```diff
@@ -217,6 +217,30 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  static int no_cloexec_support;
+  int fd;
+
+#ifdef O_CLOEXEC
+  no_cloexec_support = 1;
+#endif
+
+  fd = mkstemp((char*) req->path);
+
+  if (fd == -1) {
+    return -errno;
+  }
+
+  if (no_cloexec_support == 1) {
+    if (fcntl(fd, F_SETFD, FD_CLOEXEC) == -1) {
+      return -errno;
```

Comment:
Thanks for bearing with me, I'll fix these tonight if I can. Alon with using Ben's suggestion.